### PR TITLE
sweet extensions go.NetId() fix

### PR DESCRIPTION
**Pull request by pyrosshade**
go.NetId() can now retreive matrix netIdentities again
~~Removes unreachable return statement~~

### CODE REVIEW

File: UnityProject/Assets/Scripts/Player/PlayerSync.cs [hunk 1]
```diff
@@ -473,10 +473,6 @@ private void UpdateMe()
 
 	private void Synchronize()
 	{
-		if (isLocalPlayer && GameData.IsHeadlessServer)
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Player/PlayerSync.cs]: Technically this can trigger on the headless server,m idk how it handles its own playerr

Comment (pyrosshade) [reply UnityProject/Assets/Scripts/Player/PlayerSync.cs]: I'm under the impression that headless is server only? There shouldn't be a localplayer for something that is not a client.

**Comment by corp-0** [comment]
changelog is meant for the final user, both of these entries doesn't mean much for them.

File: UnityProject/Assets/Scripts/Util/SweetExtensions.cs [hunk 2]
```diff
@@ -82,7 +82,11 @@ public static T GetRandom<T>(this List<T> list)
 
 	public static uint NetId(this GameObject go)
 	{
-		return go ? go.GetComponent<NetworkIdentity>().netId : global::NetId.Invalid; //maxValue is invalid (see NetId.cs)
+		return go == null
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Util/SweetExtensions.cs]: I like ternaries but in this case the nested ternaries might be hard to read for people with less experience. I'd suggest using a simple if statement in this case.

File: UnityProject/Assets/Scripts/Util/SweetExtensions.cs [hunk 3]
```diff
@@ -82,7 +82,11 @@ public static T GetRandom<T>(this List<T> list)
...
+			? global::NetId.Invalid
+			: go.GetComponent<Matrix>()
+				? go.GetComponentInChildren<NetworkIdentity>().netId
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Util/SweetExtensions.cs]: Splitting it out into If statements would be better like Gilles said and then do TryGetComponent<Matrix>(out var matrix)

and then returning matrix.NetworkedMatrix.MatrixSync.netId

File: UnityProject/Assets/Scripts/Player/PlayerSync.cs (hunk 1 above)
Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Player/PlayerSync.cs]: Yeah, i dont think this check was ever true. So removing it wouldnt change anything. But just making it a Headless check (if that was the original intension) could break stuff, so would need testing.

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Player/PlayerSync.cs]: image

i donno maybe we should keep it in. Might be some weird interaction with headless

**Comment by pyrosshade** [comment]
> 
> 
> changelog is meant for the final user, both of these entries doesn't mean much for them.

gotcha, changed.

**Pull request merged**